tests/optim/models/test_models.py

```diff
@@ -1,28 +1,77 @@
 #!/usr/bin/env python3
 import unittest
+from typing import List
 
 import torch
 
 from captum.optim._models.inception_v1 import googlenet
+from captum.optim._utils.models import (
+    AvgPool2dConstrained,
+    RedirectedReluLayer,
+    ReluLayer,
+    SkipLayer,
+)
 from tests.helpers.basic import BaseTest, assertTensorAlmostEqual
 
 
+def _check_layer_in_model(self, model, layer) -> None:
+    in_model: List = []
+
+    def check_for_layer_in_model(model, layer, in_model: List) -> None:
+        for name, child in model._modules.items():
+            if child is not None:
+                if isinstance(child, layer):

```

Comment:
Could we have something like this instead?

```
    def check_for_layer_in_model(model, layer, in_model: List) -> None:
        for name, child in model._modules.items():
            if child is not None:
                if isinstance(child, layer):
                    return
                check_for_layer_in_model(child, layer, in_model)

    check_for_layer_in_model(model, layer, in_model)
    # If we reach here then there was no True 
    self.assertTrue(False)
```

Reply:
@NarineK Is your code meant to return True after finding the first layer, or else it returns False?

Reply:
yes, it basically passes the assert and returns since the function doesn't return anything I didn't return anything otherwise if we reach `self.assertTrue(False)` it should always fail, basically your original any has all false values

Reply:
Okay, I've replaced the function with your improved version!

Reply:
So your version had an error, and I realized my version could also optional check for a specified number of layers. I reverted back to my function and added the addition parameter. The new parameter is not used at the moment, but in the future it can be used to check that all layers that should have been replaced, were replaced.

Reply:
I see - what was the error ?

Reply:
@NarineK The inner function only returns to the outer function, and not the test function. So, it always fails.

Reply:
Ah, sorry, I missed the outer function.
Something like this might suffice too. 
```
def _check_layer_in_model(self, model, layer) -> None:
    def check_for_layer_in_model(model, layer, in_model: List) -> bool:
        for name, child in model._modules.items():
            if child is not None:
                if isinstance(child, layer):
                    return True
                check_for_layer_in_model(child, layer, in_model)
        return False

    self.assertTrue(check_for_layer_in_model(model, layer, in_model))

```

Reply:
@NarineK Thank you!

Reply:
@ProGamerGov, you might need the return there:
https://github.com/pytorch/captum/blob/ba076856d14a4c85f5903e371b272ac20b293c07/tests/optim/models/test_models.py#L27
I didn't test locally

Reply:
@NarineK I tested locally before adding it!

Reply:
I think in this case it works because it doesn't go to next recursive call.
I've created this simple example:
```
def a():
    i = 0
    def b():
         nonlocal i
         i += 1 
         if i < 6:
            if i == 4:
                return True
            # doesn't return b 
            b()
         return False

    print(b())    

a()
# Always return False
```